payjoin-cli/src/app/config.rs

```diff
@@ -68,16 +68,24 @@ impl AppConfig {
 
         let builder = match matches.subcommand() {
             Some(("send", _)) => builder,
-            Some(("receive", matches)) => builder
-                .set_override_option(
-                    "pj_host",
-                    matches.get_one::<String>("port").map(|port| format!("0.0.0.0:{}", port)),
-                )?
-                .set_override_option(
-                    "pj_endpoint",
-                    matches.get_one::<Url>("endpoint").map(|s| s.as_str()),
-                )?
-                .set_override_option("sub_only", matches.get_one::<bool>("sub_only").copied())?,
+            Some(("receive", matches)) => {
+                #[cfg(not(feature = "v2"))]
+                let builder = {
+                    let port = matches
+                        .get_one::<String>("port")
+                        .map(|port| port.parse::<u16>())
+                        .transpose()
+                        .expect("port argument must be a valid number");
```

Comment:
Perhaps add quotes for absolute clarity on what the name of the argument should be?
```suggestion
                        .expect("\"port\" argument must be a valid number");
```

Reply:
Nice catch, this shouldn't be expect either. Now errors look like this

```
Error: "port" argument must be a valid number

Caused by:
    invalid digit found in string```